src/Combinatorics/Matroids/properties.jl

```diff
@@ -0,0 +1,983 @@
+export
+    isomorphism, size_groundset,
+    bases, nonbases, circuits, hyperplanes, flats, cyclic_flats, closure, 
+    rank, nullity, 
+    fundamental_circuit, fundamental_cocircuit,
+    spanning_sets, independent_sets,
+    cobases, cocircuits, cohyperplanes, corank,
+    is_clutter,
+    is_regular, is_binary, is_ternary,
+    n_connected_components, connected_components, is_connected,
+    loops, coloops, is_loopless, is_coloopless, is_simple, direct_sum_components,
+    connectivity_function, is_vertical_k_separation, is_k_separation,
+    vertical_connectivity, girth, tutte_connectivity,
+    tutte_polynomial, characteristic_polynomial, charpoly, reduced_characteristic_polynomial,
+    revlex_basis_encoding
+
+################################################################################
+##  Properties and basic functions
+################################################################################
+
+function isomorphism(M::Matroid, gs::AbstractVector)
+    if lenght(M.groundset)!=length(gs)
+        error("Sets of different size")
+    end
+    gs2num = Dict{Any,Int}()
+    i = 1
+    for elem in gs
+        gs2num[elem] = i
+        i+=1
+    end
+    return Matroid(M.pm_matroid,gs,gs2num)
+end
+
+size_groundset(M::Matroid) = length(M.groundset)
+
+@doc Markdown.doc"""
+    bases(M::matroid)
+
+Return the list of bases of the matroid `M`.
+
+# Example
+```jldoctest
+julia> bases(uniform_matroid(2, 3))
+3-element Vector{Vector{Int64}}:
+ [1, 2]
+ [1, 3]
+ [2, 3]
+```
+"""
+bases(M::Matroid) = [[M.groundset[i+1] for i in sort(collect(C))] for C in Vector{Set{Int}}(M.pm_matroid.BASES)]
+
+@doc Markdown.doc"""
+    nonbases(M::matroid)
+
+Return the list of nonbases of the matroid `M`.
+
+# Example
+```jldoctest
+julia> nonbases(fano_matroid())
+7-element Vector{Any}:
+ [1, 6, 7]
+ [2, 5, 7]
+ [3, 4, 7]
+ [3, 5, 6]
+ [2, 4, 6]
+ [1, 4, 5]
+ [1, 2, 3]
+
+```
+"""
+function nonbases(M::Matroid)
+    Nonbases = []
+    Bases = bases(M)
+    for set in Oscar.Hecke.subsets(Vector(1:length(M.groundset)),rank(M))
+        if !(set in Bases)
+            push!(Nonbases, set)
+        end
+    end
```

Comment:
Can this block not be refactored as some variant of `filter`?
```
julia> v = [i for i in -3:3]
7-element Vector{Int64}:
 -3
 -2
 -1
  0
  1
  2
  3

julia> filter(i -> i%2==0, v)
3-element Vector{Int64}:
 -2
  0
  2
```

Reply:
Done.